code/client/src/core/modules/vehicle.cpp

```diff
@@ -37,27 +37,46 @@ namespace MafiaMP::Core::Modules {
-                    metadata.brake             = vehicle->GetBrake();
-                    metadata.gear              = car->GetGear();
-                    metadata.handbrake         = vehicle->GetHandbrake();
-                    metadata.hornState         = vehicle->GetHorn();
-                    metadata.power             = vehicle->GetPower();
-                    metadata.radioId           = vehicle->GetRadioStation();
-                    metadata.radioState        = vehicle->IsRadioOn();
-                    metadata.sirenState        = vehicle->GetSiren();
-                    metadata.steer             = vehicle->GetSteer();
-                    metadata.velocity          = {vehicleVelocity.x, vehicleVelocity.y, vehicleVelocity.z};
+
+                    SDK::ue::sys::math::C_Vector4 colorPrimary, colorSecondary;
+                    vehicle->GetVehicleColor(&colorPrimary, &colorSecondary);
+
+                    SDK::ue::sys::math::C_Vector4 rimColor, tireColor;
+                    vehicle->GetWheelColor(&rimColor, &tireColor);
+
+                    SDK::ue::sys::math::C_Vector4 windowTint = vehicle->GetWindowTintColor();
+
+                    tr.pos = {carPos.x, carPos.y, carPos.z};
+                    tr.rot = {carRot.w, carRot.x, carRot.y, carRot.z};
+
+                    metadata.angularVelocity = {vehicleAngularVelocity.x, vehicleAngularVelocity.y, vehicleAngularVelocity.z};
+                    metadata.beaconLightsOn  = vehicle->GetBeaconLightsOn();
+                    metadata.brake           = vehicle->GetBrake();
+                    metadata.colorPrimary    = {colorPrimary.r, colorPrimary.g, colorPrimary.b, colorPrimary.a};
+                    metadata.colorSecondary  = {colorSecondary.r, colorSecondary.g, colorSecondary.b, colorSecondary.a};
+                    metadata.dirt            = vehicle->GetVehicleDirty();
+                    metadata.fuel            = car->GetActualFuel();
+                    metadata.gear            = car->GetGear();
+                    metadata.handbrake       = vehicle->GetHandbrake();
+                    metadata.hornOn          = vehicle->GetHorn();
+                    metadata.power           = vehicle->GetPower();
+                    metadata.radioOn         = vehicle->IsRadioOn();
+                    metadata.radioStationId  = vehicle->GetRadioStation();
+                    metadata.rimColor        = {rimColor.r, rimColor.g, rimColor.b, rimColor.a};
+                    metadata.rust            = vehicle->GetVehicleRust();
+                    metadata.sirenOn         = vehicle->GetSiren();
+                    metadata.steer           = vehicle->GetSteer();
+                    metadata.tireColor       = {tireColor.r, tireColor.g, tireColor.b, tireColor.a};
+                    metadata.velocity        = {vehicleVelocity.x, vehicleVelocity.y, vehicleVelocity.z};
+                    metadata.windowTint      = {windowTint.r, windowTint.g, windowTint.b, windowTint.a};
```

Comment:
@zpl-zak ok with this? Makes a lot of data to send in the frequent update payload

Reply:
it's fine for now, we will turn these into delta patches later on eventually (bs.SerializeDelta)